### PR TITLE
feat(notifications): Bulk action Delete endpoint

### PR DESCRIPTION
**Pull request by dmyios**
## Description

Implementation of Bulk action Delete endpoint

`POST /configuration/notifications/_delete`

**Fixes** # MON-19634

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Notification/Infrastructure/Repository/DbWriteNotificationRepository.php [hunk 1]
```diff
@@ -132,17 +133,22 @@ public function addUsers(int $notificationId, array $userIds): void
...
-        $statement = $this->db->prepare($this->translateDbName($request));
-        $statement->bindValue(':notification_id', $notificationId, \PDO::PARAM_INT);
-        $statement->execute();
+        try {
+            $statement = $this->db->prepare($this->translateDbName($request));
+            $statement->bindValue(':notification_id', $notificationId, \PDO::PARAM_INT);
+            $statement->execute();
+
+            return $statement->rowCount();
+        } catch (\Exception $ex) {
```
Comment (dmyios) [inline centreon/src/Core/Notification/Infrastructure/Repository/DbWriteNotificationRepository.php:150]:
```suggestion
        } catch (\PDOException $ex) {
```

File: centreon/src/Core/Notification/Application/UseCase/DeleteNotifications/DeleteNotifications.php [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            if ($this->contactCanExecuteUseCase()) {
+                $response = new DeleteNotificationsResponse();
+                $results = [];
+
+                foreach ($request->ids as $notificationId) {
+                    try {
+                        $statusResponse = $this->deleteNotification($notificationId);
+                        $responseStatusDto = $this->createStatusResonseDto($statusResponse, $notificationId);
+                        $results[] = $responseStatusDto;
+                    } catch (RepositoryException $ex) {
```
Comment (dmyios) [inline centreon/src/Core/Notification/Application/UseCase/DeleteNotifications/DeleteNotifications.php:74]:
```suggestion
                    } catch (\Throwable $ex) {
```

File: centreon/src/Core/Notification/Infrastructure/Repository/DbWriteNotificationRepository.php [hunk 3]
```diff
@@ -132,17 +133,22 @@ public function addUsers(int $notificationId, array $userIds): void
...
+            throw new RepositoryException($ex->getMessage());
+        }
```
Comment (dmyios) [inline centreon/src/Core/Notification/Infrastructure/Repository/DbWriteNotificationRepository.php:152]: Remove try...catch block from repository method

File: centreon/src/Core/Infrastructure/Common/Presenter/JsonFormatter.php [hunk 4]
```diff
@@ -150,7 +153,9 @@ private function generateJsonResponse(mixed $data, int $code, array $headers): J
         if (is_object($data)) {
             if ($data instanceof \Generator) {
                 $data = iterator_to_array($data);
-            } elseif ($data instanceof CreatedResponse) {
+            } else if ($data instanceof CreatedResponse) {
```
Comment (jeremyjaouen) [inline centreon/src/Core/Infrastructure/Common/Presenter/JsonFormatter.php:156]:
```suggestion
            } else if ($data instanceof CreatedResponse || $data instanceof MultiStatusResponse) {
```

**Review by jeremyjaouen — APPROVED** [review]

**Review by wtermellil — APPROVED** [review]

**Pull request merged**